drivers/adc/adc_ads1119.c

```diff
@@ -0,0 +1,522 @@
+		return -ENOTSUP;
+	}
+
+	if (sequence->oversampling) {
+		return -ENOTSUP;
+	}
+
+	return ads1119_validate_buffer_size(sequence);
+}
+
+static void adc_context_update_buffer_pointer(struct adc_context *ctx,
+					      bool repeat_sampling)
+{
+	struct ads1119_data *data = CONTAINER_OF(ctx,
+						 struct ads1119_data,
+						 ctx);
+
+	if (repeat_sampling) {
+		data->buffer = data->buffer_ptr;
+	}
+}
+
+static void adc_context_start_sampling(struct adc_context *ctx)
+{
+	struct ads1119_data *data = CONTAINER_OF(ctx,
+						 struct ads1119_data, ctx);
+
+	data->buffer_ptr = data->buffer;
+	k_sem_give(&data->acq_sem);
+}
+
+static int ads1119_adc_start_read(const struct device *dev,
+				  const struct adc_sequence *sequence,
+				  bool wait)
+{
+	int rc;
+	struct ads1119_data *data = dev->data;
+
+	rc = ads1119_validate_sequence(sequence);
+	if (rc != 0) {
+		return rc;
+	}
+
+	data->buffer = sequence->buffer;
+
+	adc_context_start_read(&data->ctx, sequence);
+
+	if (wait) {
+		rc = adc_context_wait_for_completion(&data->ctx);
+	}
+	return rc;
+}
+
+static int ads1119_adc_perform_read(const struct device *dev)
+{
+	int rc;
+	struct ads1119_data *data = dev->data;
+
+	k_sem_take(&data->acq_sem, K_FOREVER);
+
+	rc = ads1119_send_start_read(dev);
+	if (rc) {
+		adc_context_complete(&data->ctx, rc);
+		return rc;
+	}
+
+	rc = ads1119_wait_data_ready(dev);
+	if (rc != 0) {
+		adc_context_complete(&data->ctx, rc);
+		return rc;
+	}
+
+	rc = ads1119_read_sample(dev, data->buffer);
+	if (rc != 0) {
+		adc_context_complete(&data->ctx, rc);
+		return rc;
+	}
+	data->buffer++;
+
+	adc_context_on_sampling_done(&data->ctx, dev);
+
+	return rc;
+}
+
+#if CONFIG_ADC_ASYNC
+static int ads1119_read(const struct device *dev,
+			const struct adc_sequence *sequence)
+{
+	int rc;
+	struct ads1119_data *data = dev->data;
+
+	adc_context_lock(&data->ctx, false, NULL);
+	rc = ads1119_adc_start_read(dev, sequence, true);
+	adc_context_release(&data->ctx, rc);
+
+	return rc;
+}
+#else
+static int ads1119_read(const struct device *dev,
+			const struct adc_sequence *sequence)
+{
+	int rc;
+	struct ads1119_data *data = dev->data;
+
+	adc_context_lock(&data->ctx, false, NULL);
+	rc = ads1119_adc_start_read(dev, sequence, false);
+
+
+	while (rc == 0 && k_sem_count_get(&data->ctx.sync) == 0) {
+		rc = ads1119_adc_perform_read(dev);
+	}
+	k_sem_take(&data->ctx.sync, K_FOREVER);
+	adc_context_release(&data->ctx, rc);
+	return rc;
+}
+#endif
+
+#if CONFIG_ADC_ASYNC
+static void ads1119_acquisition_thread(struct device *dev)
+{
+	while (true) {
+		ads1119_adc_perform_read(dev);
+	}
+}
+#endif
+
+static int ads1119_init(const struct device *dev)
+{
+	int rc;
+	uint8_t status;
+	const struct ads1119_config *config = dev->config;
+	struct ads1119_data *data = dev->data;
+
+	adc_context_init(&data->ctx);
+
+	k_sem_init(&data->acq_sem, 0, 1);
+
+	if (!device_is_ready(config->i2c_dev)) {
+		return -ENODEV;
+	}
+
+	rc = ads1119_read_reg(dev, ADS1119_REG_STATUS, &status);
```

Comment:
is this used anywhere? If not, remove it

Reply:
It is used to check that the adc can be accessed